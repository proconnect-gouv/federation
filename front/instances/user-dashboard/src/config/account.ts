import type { AccountConfig } from '@fc/account';

export const Account: AccountConfig = {
  endpoints: {
    login: '/api/redirect-to-idp',
    logout: '/api/logout',
    me: '/api/me',
  },
};
