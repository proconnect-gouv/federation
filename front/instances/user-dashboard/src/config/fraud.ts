import type { FraudConfigInterface } from '@fc/user-dashboard';

export const Fraud: FraudConfigInterface = {
  apiRouteFraudForm: `/api/fraud-form`,
  fraudSupportFormPathname: '/usurpation',
  fraudSurveyUrl: 'https://aide.franceconnect.gouv.fr/erreurs/signalement/etape-1/',
  supportFormUrl: 'https://app.franceconnect.gouv.fr/support/formulaire',
  surveyOriginQueryParam: 'fraudSurveyOrigin',
};
