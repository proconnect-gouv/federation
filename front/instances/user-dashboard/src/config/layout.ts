/* istanbul ignore file */

// declarative file
import { LogoFranceConnect } from '@fc/assets';
import type { LayoutConfig } from '@fc/layout';

export const Layout: LayoutConfig = {
  features: {
    showLicence: false,
    showServiceTitle: false,
  },
  footer: {
    description:
      'FranceConnect est le dispositif d’identification conçu par l’État pour vous faciliter l’accès à vos services en ligne. En utilisant un de vos comptes déjà existants, vous pourrez vous connecter de façon sécurisée à plus de 1000 services sans créer de nouveau mot de passe.',
    links: [
      {
        href: 'https://www.legifrance.gouv.fr',
        label: 'legifrance.gouv.fr',
        title: 'Accéder au site legifrance.gouv.fr nouvelle fenêtre',
      },
      {
        href: 'https://www.info.gouv.fr',
        label: 'info.gouv.fr',
        title: 'Accéder au site info.gouv.fr nouvelle fenêtre',
      },
      {
        href: 'https://www.service-public.fr/',
        label: 'service-public.fr',
        title: 'Accéder au site service-public.fr nouvelle fenêtre',
      },
      {
        href: 'https://data.gouv.fr',
        label: 'data.gouv.fr',
        title: 'Accéder au site data.gouv.fr nouvelle fenêtre',
      },
    ],
    navigation: [
      {
        href: 'https://franceconnect.gouv.fr/plan-site',
        label: 'Plan du site',
      },
      {
        href: 'https://franceconnect.gouv.fr/accessibilite',
        label: 'Accessibilité : non conforme',
      },
      {
        href: 'https://franceconnect.gouv.fr/mentions-legales',
        label: 'Mentions légales',
      },
      {
        href: 'https://franceconnect.gouv.fr/cgu',
        label: 'Données personnelles',
      },
      {
        href: 'https://franceconnect.gouv.fr/cgu',
        label: 'Gestion des cookies',
      },
    ],
  },
  navigation: [
    {
      href: '/history',
      label: 'Mon historique de connexion',
    },
    {
      href: '/preferences',
      label: 'Gérer mes accès',
    },
    {
      href: '/fraud/form',
      label: 'Signaler une usurpation',
    },
  ],
  service: {
    homepage: '/',
    logo: LogoFranceConnect,
  },
};
