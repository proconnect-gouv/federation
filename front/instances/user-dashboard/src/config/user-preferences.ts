/* istanbul ignore file */

// declarative file
import { UserPreferencesConfig } from '@fc/user-preferences';

const API_BASE_URL = '/api';

export const UserPreferences: UserPreferencesConfig = {
  API_ROUTE_CSRF_TOKEN: `${API_BASE_URL}/csrf-token`,
  API_ROUTE_USER_PREFERENCES: `${API_BASE_URL}/user-preferences`,
};
