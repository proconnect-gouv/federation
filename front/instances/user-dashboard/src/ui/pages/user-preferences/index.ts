export * from './user-preferences.page';
