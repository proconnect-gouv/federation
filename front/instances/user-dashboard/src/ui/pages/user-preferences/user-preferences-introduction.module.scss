.title {
  color: var(--text-active-blue-france);
}

.tutoriel {
  width: 100%;

  @media screen and (min-width: 992px) {
    width: 350px;
  }

  > img {
    border: 2px solid var(--border-active-blue-france);
    border-radius: 8px;
  }
}
