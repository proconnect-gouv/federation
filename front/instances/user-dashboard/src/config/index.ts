/* istanbul ignore file */

// declarative file
import { Account } from './account';
import { Fraud } from './fraud';
import { HttpClient } from './http-client';
import { Layout } from './layout';
import { Tracks } from './tracks';
import { UserPreferences } from './user-preferences';

export const AppConfig = {
  Account,
  Fraud,
  HttpClient,
  Layout,
  Tracks,
  UserPreferences,
};
