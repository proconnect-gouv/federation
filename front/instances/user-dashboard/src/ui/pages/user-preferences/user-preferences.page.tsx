import classnames from 'classnames';
import React from 'react';
import { Helmet } from 'react-helmet-async';

import { useStylesQuery, useStylesVariables } from '@fc/styles';
import { UserPreferencesComponent } from '@fc/user-preferences';

import { UserPreferencesIntroductionComponent } from '../../components';

export const UserPreferencesPage = React.memo(() => {
  const [breakpointLg] = useStylesVariables(['breakpoint-lg']);

  const gtDesktop = useStylesQuery({ minWidth: breakpointLg });

  return (
    <React.Fragment>
      <Helmet>
        <title>Mon tableau de bord - Mes Acc√®s</title>
      </Helmet>
      <div
        className={classnames('fr-m-auto fr-px-2w', {
          // Class CSS
          // eslint-disable-next-line @typescript-eslint/naming-convention
          'fr-mt-5w': !gtDesktop,
          // Class CSS
          // eslint-disable-next-line @typescript-eslint/naming-convention
          'fr-mt-8w': gtDesktop,
        })}
        id="page-container">
        <UserPreferencesComponent />
        <UserPreferencesIntroductionComponent />
      </div>
    </React.Fragment>
  );
});

UserPreferencesPage.displayName = 'UserPreferencesPage';
