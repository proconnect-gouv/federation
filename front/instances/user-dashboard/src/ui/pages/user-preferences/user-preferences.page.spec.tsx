import { render } from '@testing-library/react';
import { useMediaQuery } from 'react-responsive';

import { UserPreferencesComponent } from '@fc/user-preferences';

import { AppConfig } from '../../../config';
import { UserPreferencesPage } from './user-preferences.page';

jest.mock('react-responsive');
jest.mock('@fc/user-preferences');

describe('UserPreferencesPage', () => {
  it('should match the snapshot, display into a desktop viewport', () => {
    // given
    jest.mocked(useMediaQuery).mockReturnValueOnce(true);

    // when
    const { container } = render(<UserPreferencesPage />);

    // then
    expect(container).toMatchSnapshot();
  });

  it('should match the snapshot, display into a mobile viewport', () => {
    // given
    jest.mocked(useMediaQuery).mockReturnValueOnce(false);

    // when
    const { container } = render(<UserPreferencesPage />);

    // then
    expect(container).toMatchSnapshot();
  });

  it('should call UserPreferencesComponent with props', () => {
    // when
    render(<UserPreferencesPage />);

    // then
    expect(UserPreferencesComponent).toHaveBeenCalledOnce();
    expect(UserPreferencesComponent).toHaveBeenCalledWith(
      { options: AppConfig.UserPreferences },
      {},
    );
  });
});
