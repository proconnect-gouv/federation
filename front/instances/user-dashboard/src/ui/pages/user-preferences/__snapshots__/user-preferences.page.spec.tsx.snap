// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserPreferencesPage should match the snapshot, display into a desktop viewport 1`] = `
<div>
  <div
    data-mockid="Helmet"
  >
    <title>
      Mon tableau de bord - Mes Accès
    </title>
  </div>
  <div
    class="fr-m-auto fr-px-2w fr-mt-8w"
    id="page-container"
  >
    <div>
      UserPreferencesComponent
    </div>
    <div>
      UserPreferencesIntroductionComponent
    </div>
  </div>
</div>
`;

exports[`UserPreferencesPage should match the snapshot, display into a mobile viewport 1`] = `
<div>
  <div
    data-mockid="Helmet"
  >
    <title>
      Mon tableau de bord - Mes Accès
    </title>
  </div>
  <div
    class="fr-m-auto fr-px-2w fr-mt-5w"
    id="page-container"
  >
    <div>
      UserPreferencesComponent
    </div>
    <div>
      UserPreferencesIntroductionComponent
    </div>
  </div>
</div>
`;
